src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift

```diff
@@ -56,7 +56,12 @@ extension RiskProvidingConfiguration {
 		return currentDate < exposureDetectionValidUntil(lastExposureDetectionDate: lastExposureDetectionDate)
 	}
 
-	func shouldPerformExposureDetection(lastExposureDetectionDate: Date?, currentDate: Date = Date()) -> Bool {
+	func shouldPerformExposureDetection(activeTracingHours: Int, lastExposureDetectionDate: Date?, currentDate: Date = Date()) -> Bool {
+		// Don't allow exposure detection within the first frame of exposureDetectionInterval
+		if activeTracingHours < ((exposureDetectionInterval.day ?? 1) * 24) {
+			return false
+		}
+
 		if let lastExposureDetectionDate = lastExposureDetectionDate, lastExposureDetectionDate > currentDate {
```

Comment:
in which case could the lastExposureDetectionDate lie in the future? Sounds like it might be worth checking out why this can lie in the future, and find the method / the location where this is from?

Reply:
If you manually change the current date in the system settings because you don't want to wait 24 hours until the next risk calculation and then go back to your actual date